### DISCUSSION TITLE
关于PCL2到底用的是什么下载源

### DISCUSSION THREAD

**Discussion by E0x72-21** [Meloong-Git/PCL#4820] [提问] [answered]
这里的选项只写了版本列表获取方式，但最后的下载到底是从哪下的？
![image](https://github.com/user-attachments/assets/3ffa6aa1-2d8f-4e90-af92-1c14cd20e9d2)
而且我现在不在大陆，学校的校园网似乎墙PCL2所用的下载源，上图这个选项所有的都试过了不好使（已知校园网环境下modrinth app和官启的下载/安装一切正常）所以有没有办法更改PCL2的下载源以应对国外的网络环境？

**Comment by WForst-Breeze** [comment]
PCL 当前尚不支持修改下载源

在 MCBBS 关站前，下载源是这样尝试的： https://github.com/Hex-Dragon/PCL2/issues/3213#issuecomment-1913536797 ，现在应该也是差不多的？

**Reply by E0x72-21** [reply]
那为什么这个多语言版本在我校园网也下不动游戏？（https://github.com/Hex-Dragon/PCL2/discussions/4580） 这玩意不应该优先走官方或者不是国内的源吗？

**Reply by tangge233** [reply]
> 那为什么这个多语言版本在我校园网也下不动游戏？（https://github.com/Hex-Dragon/PCL2/discussions/4580） 这玩意不应该优先走官方或者不是国内的源吗？

那个版本我调整了版本列表获取默认设置，以为在走官方源了。回头我改改（

**Reply by E0x72-21** [reply]
> > 那为什么这个多语言版本在我校园网也下不动游戏？（[https://github.com/Hex-Dragon/PCL2/discussions/4580）](https://github.com/Hex-Dragon/PCL2/discussions/4580%EF%BC%89) 这玩意不应该优先走官方或者不是国内的源吗？
> 
> 那个版本我调整了版本列表获取默认设置以为在走官方源了，回头我改改（

不是版本列表获取源，是下载源啊（恼

**Comment by tangge233** [comment]
BMCLAPI 封禁了一些 IP 来缓解网络攻击，所以很可能不是你的校园网墙了镜像源（悲

**Reply by E0x72-21** [reply]
不清楚，我校园网都墙qq百度知乎之类的(b站不知道为什么幸免遇难)，甚至当地的家宽公网ip也墙，难说会不会墙镜像源

**Reply by allMagicNB** [reply]
BMACAPI 是什么？

**Reply by Joker2184** [reply]
> BMACAPI 是什么？

就BMCL镜像源啊）

**Reply by 3gf8jv4dv** [reply]
> 就BMCL镜像源啊）

This is because tangge233 made a typo before and allMagic just reminded him.
这是因为之前 tangge233 打错字了，allMagic 只是提醒他。

**Comment by LTCatt** [comment]
如果下载不动，可以先发个日志过来看看，PCL 文件夹下面的 Log1~5.txt

**Reply by E0x72-21** [reply]
[Log1.txt](https://github.com/user-attachments/files/17150914/Log1.txt) 或者 https://mclo.gs/JNZ3neW
我自己也翻了一下似乎全是走bmclapi下载源的卡住了

**Reply by LTCatt** [reply, marked as answer]
PCL 会在第一轮尝试走 BMCLAPI，似乎你在第一轮尝试结束前就关了，理论上等得足够久，把三千多个文件都失败一次之后，是能下载的 Orz……

**Reply by E0x72-21** [reply]
> PCL 会在第一轮尝试走 BMCLAPI，似乎你在第一轮尝试结束前就关了，理论上等得足够久，把三千多个文件都失败一次之后，是能下载的 Orz……

就不能整一个能让用户手动改下载源的选项吗（

**Reply by Deep-Dark-Forest** [reply]
> > PCL 会在第一轮尝试走 BMCLAPI，似乎你在第一轮尝试结束前就关了，理论上等得足够久，把三千多个文件都失败一次之后，是能下载的 Orz……
> 
> 就不能整一个能让用户手动改下载源的选项吗（

#3466

**Reply by tangge233** [reply]
本地化这边打算让非简中用户优先尝试官方源（

**Comment by hekuo5310** [comment]
原来是ban了自定义啊.......
那看来自己的源只能靠代理+改hosts了
